dataplane/windows/policysets/policysets.go

```diff
@@ -361,6 +361,71 @@ func (s *PolicySets) protoRuleToHnsRules(policyId string, pRule *proto.Rule, isI
 		logCxt.WithField("action", ruleCopy.Action).Panic("Unknown rule action")
 	}
 
+	//
+	// DstIpPort sets - these cannot co-exist with other fields, so do them first and short-circuit if set.
+	//
+	if len(ruleCopy.DstIpPortSetIds) > 0 {
+		ipsetMembers, err := s.getIPSetAddresses(ruleCopy.DstIpPortSetIds)
+		if err != nil {
+			logCxt.Warn("DstIpPortSetIds could not be resolved, rule will be skipped")
```

Comment:
If a policy reference a service but all backend endpoints are degraded, will we reach here?

Reply:
Nope, it will return `[]string{}, nil` and we'll properly generate just the default deny rules.

I added a UT for this case according to your recommendation :)